internal/gatewayapi/clienttrafficpolicy.go

```diff
@@ -335,3 +373,64 @@ func translateListenerProxyProtocol(enableProxyProtocol *bool, httpIR *ir.HTTPLi
 		httpIR.EnableProxyProtocol = true
 	}
 }
+
+func translateListenerTLSParameters(tlsParams *egv1a1.TLSSettings, httpIR *ir.HTTPListener) error {
+	// Return early if not set
+	if tlsParams == nil {
+		return nil
+	}
+	// Return if this listener isn't a TLS listener. There has to be
+	// at least one certificate defined, which would cause httpIR to
+	// have a TLS structure.
+	if httpIR.TLS == nil {
+		return nil
+	}
+	var errs error
+	if tlsParams.Version != nil {
+		var minVersion, maxVersion int
+		var err error
+		if minVersion, err = tlsVersionToNumericVersion(tlsParams.Version.Min); err != nil {
```

Comment:
This seems to imply that settings like this:

```yaml
version:
  min: TLSv1_2
```

will not work, since it seems that the default for `max` is `TLS_Auto`.

I think that "auto" should mean "the earliest supported version" for minimum and "the latest supported version" for maximum. (While we're at it, I also don't see any value in forcing people to type "TLS_" everywhere: just let the versions be strings like "v1.3", and "auto".)

Reply:
This was resolved by adopting defaults of minimum TLS v1.2 and maximum TLS v1.3.

The validation that `min <= max` was implemented as a CEL rule.